docs/proposals/disk-affinity.md

```diff
@@ -1,40 +1,93 @@
 
-antiaffinity.volumes.csi-baremetal.dell.com/labels: "${list of the labels}" - do not mount PVs on the disks that already have some PVs from other specific pods (specified by labels). \
-antiaffinity.volumes.csi-baremetal.dell.com/volumes: "[pv-1, pv-2], [pv-1, pv3], ..." - specify which pvs should not be placed on the same disks.
+As we have already described in previous section (Rationale for annotations usage) - we are going to use annotations for use cases supporting.
+In order to specify how client is going to use disk/affinity the following pod annotation is introduced:
+```yaml
+affinity.volumes.csi-baremetal.dell.com/types: ${list of specified affinity types}
+antiaffinity.volumes.csi-baremetal.dell.com/types: ${list of specified antiaffinity types}
+```
+The list of supported **_affinity_** types to use in the above annotation:
+* ```pod-bound-(required|preferred)``` - mount PVs for the same pod on the same disk, depending on the suffix - the behaviour is required/preferred;
+* ```dedicated-(required|preferred)``` - place volumes on the certain disks, which were dedicated for this application, depending on the suffix - the behaviour is required/preferred.
 
-If we want to dedicate drive/lvg for the certain application, then we should place label on it's ac: \
-affinity.csi-baremetal.dell.com/taint: nginx
+The list of supported **_antiaffinity_** types to use in the above annotation:
+* ```pod-label-(required|preferred)``` - specify pods by their labels, which should not use the same drives as the current one, depending on the suffix - the behaviour is required/preferred;
+
+#### Mount PVs for the same pod on the same disk
+For supporting mounting PVs for the same pod on the same disk the following annotation should be specified on the specific pod before its creation:
+```yaml
+affinity.volumes.csi-baremetal.dell.com/types: <pod-bound-required|pod-bound-preferred>
+```
+
+#### Do not mount PVs on the disks that already have some PVs from other specific pods
+For ignoring disks, that already have some PVs from other specific pods the following annotation should be specified on our application (pod) before its creation:
+```yaml
+antiaffinity.volumes.csi-baremetal.dell.com/types: <pod-label-required|pod-label-preferred>
+```
+Also, the following annotation should be specified on the same pod as well:
+```yaml
+antiaffinity.volumes.csi-baremetal.dell.com/labels: ${pod-label1}:${pod-namespace1},...,${pod-labelN}:${pod-namespaceM}
+```
+In the above annotation we should specify labels of specific pods, whose disks in usage we do not want to consume by PVs of our application (pod).
+
+#### Directly dedicate disks for specific pods
+Generally, we have two options to support directly dedicated disks (which one is preferable is the space for discussion):
+* Set the following annotation on the certain pod before its creation to request dedicated disks for it:
+  ```yaml
+  affinity.volumes.csi-baremetal.dell.com/dedicated: true
```

Comment:
Can we make `required|preferred` instead of `true`?

Reply:
In this particular case we can. Pushed the commit